IAC IAT PWM Curve addition

shiznit304 opened the pull request:
Feature Request: IAT vs IACV correction table https://github.com/noisymime/speeduino/issues/750

I decided to add a curve for IAT IAC corrections. I was going to do a 3d table, but that takes up too much space. 120 bytes vs 40 bytes. This PR only uses 9 additional bytes for IAT IAC Duty cycle table. the IAT bins is shared with airDenBins

shiznit304 commented:
@choyr feel free to try this one again. Be sure to also update the speeduino.ini file in your project

choyr commented:
Alright. On friday i will.

shiznit304 commented:
@choyr I tested the firmware with a brand new Mega2560. Uploaded firmware, new project with correct ini, and loaded a current tune. I tested by changing a variable in each page and couldn't reproduce your issue. I have had this issue with my board that's in my Miata but not really sure.

choyr commented:
Hi. Will try again, but usually when theres a big change it happens, like a whole ve table then it happens

choyr commented:
@shiznit304 hey man, if possible can you email me at choyrowel@yahoo.com?

choyr commented:
@shiznit304 hey thanks. Will try it once you did it. I believe josh was asking for it last time to be included in the official release. Its really handy, since summer is coming again. 
I pm'd you at slack, i need to ask something more technical.
Thanks a lot

henriquebelaz commented:
@shiznit304 Hey, I'd also like to thank you for this. Merged your PR with the latest commit from master and it worked flawlessly.
Even with low PID OL+CL settings, my idle would be hunting all around when IAT went up because of the OL duty cycle.
Now CL has one less variable to take into account.
Great addition!

choyr commented:
![image](https://user-images.githubusercontent.com/85299549/232238535-17820cf1-0d16-40ed-b81b-88a0c05de8ce.png)

I am getting this #include schedule_calcs.inl not found. in your version, you have  schedule_calcs.inl, schedule_calcs.cpp, schedule_calcs.h that is not there is 202207.1

do i have to manually edit the changes in 202207.1 instead?

shiznit304 commented:
I’m not too sure. It might be a library missing in your IDE.

On Sat, Apr 15, 2023 at 11:44 AM choyr ***@***.***> wrote:

> [image: image]
> <https://user-images.githubusercontent.com/85299549/232238535-17820cf1-0d16-40ed-b81b-88a0c05de8ce.png>
>
> I am getting this #include schedule_calcs.inl not found. in your version,
> you have schedule_calcs.inl, schedule_calcs.cpp, schedule_calcs.h that is
> not there is 202207.1
>
> do i have to manually edit the changes in 202207.1 instead?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noisymime/speeduino/pull/914#issuecomment-1509890232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO6IHZEHLJDS4DGFIBVQQ7LXBLF6JANCNFSM56WCAGPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

henriquebelaz commented:
> ![image](https://user-images.githubusercontent.com/85299549/232238535-17820cf1-0d16-40ed-b81b-88a0c05de8ce.png)
> 
> I am getting this #include schedule_calcs.inl not found. in your version, you have schedule_calcs.inl, schedule_calcs.cpp, schedule_calcs.h that is not there is 202207.1
> 
> do i have to manually edit the changes in 202207.1 instead?

I did the following:
Go to speeduino folder and rename "schedule_calcs.inl" to "schedule_calcs.ino"
Inside "schedule_calcs.h", remove the #include "schedule_calcs.inl" line.
It should be enough to compile and upload. Make sure schedule_calcs.ino is opened inside Arduino IDE with speeduino.ino and others.

choyr commented:
hey did it and now it compiles. will test it again next week

choyr commented:
hey man, is this firmware still with the 202207.1?

shiznit304 commented:
I don’t remember when’s the last time I updated it. It should show my
latest commits. I’ll check on this later


On Mon, Sep 4, 2023 at 5:32 PM choyr ***@***.***> wrote:

> hey man, is this firmware still with the 202207.1?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noisymime/speeduino/pull/914#issuecomment-1705741541>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO6IHZEWTMD4BA2PJZ3RP63XYZJGNANCNFSM56WCAGPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

choyr commented:
its 2023.06, will test it this weekend. thanks for updating!

shiznit304 committed:
iacIATPWM

commit fc3e581bc3a3aaab6b5e040dba0a6eb231683d04
Author: shiznit304 <62686180+shiznit304@users.noreply.github.com>
Date:   Sun Apr 7 21:21:31 2024 -0500

    iacIATPWM

choyr commented:
is this with 202402 firmware?

shiznit304 commented:
I have updated it to master as of a few days ago . It should work with
202402.

On Thu, Apr 11, 2024 at 11:36 AM PnPduino ***@***.***> wrote:

> is this with 202402 firmware?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noisymime/speeduino/pull/914#issuecomment-2050088519>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO6IHZB6WHLKIJEA7LKLMELY4232TAVCNFSM56WCAGP2U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TEMBVGAYDQOBVGE4Q>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

choyr commented:
right. will test it next week. any reason this can't be included in the master firmware version? last time i remember josh said there's a conflict or something

shiznit304 commented:
The discussions were about how closed loop should be able to take care of
this but I disagree. This addition I created helps a lot with heat soaked
starts.

On Thu, Apr 11, 2024 at 11:59 AM PnPduino ***@***.***> wrote:

> right. will test it next week. any reason this can't be included in the
> master firmware version? last time i remember josh said there's a conflict
> or something
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noisymime/speeduino/pull/914#issuecomment-2050124144>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO6IHZGKYOTBZWBA23CPTPDY426QDAVCNFSM56WCAGP2U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TEMBVGAYTENBRGQ2A>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

choyr commented:
yes I agree. from my experience, the toyota 4AGE and Honda K series do not have heat soak issues, but the Honda B/D/H series goes up to 75C IAT and it really lowers the rpm a lot that engine dies. this correction compensates for it and not the closed loop

choyr commented:
tested the latest version and there seems some issue. engine just stops and rpm is not steady

choyr commented:
hey @shiznit304  seems like this latest version does not work? i just tried it with a simulator and iacv does not change with iat change

shiznit304 commented:
@choyr I have merged the latest master into this PR

choyr commented:
@noisymime Hi Josh, this feature can be included in the main firmware update? Tested with this feature and it really idles a lot better than without it.

shiznit304 commented:
I haven't touched Speeduino for about a year. I'll see about getting my PR up to date

choyr commented:
Hello shiznit304, any idea if feature is merged to main? Possible to add this @noisymime
